src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,650 @@
+        result = self._getFromCfgSvc(firstOption, secondOption, defaultOption)
+        if result["OK"]:
+            self.proxyTimeLeftBeforeRenewal = result["Value"]
+
+        self.arcRESTTimeout = 1.0
+        variable = "ARCRESTTimeout"
+        firstOption = cfgPath("Resources", "Sites", self.grid, self.site, "CEs", self.ceHost, variable)
+        secondOption = cfgPath("Resources", "Sites", self.grid, self.site, variable)
+        defaultOption = cfgPath("Resources", "Computing" "CEDefaults", variable)
+        result = self._getFromCfgSvc(firstOption, secondOption, defaultOption)
+        if result["OK"]:
+            self.arcRESTTimeout = result["Value"]
+
+    #############################################################################
+    @property
+    def initialised(self):
+        """Check if the REST interface is available and ready, configuring it if required.
+        Hopefully this is executed just once per CE, though it should not harm to run it more often.
+        If all is well the following variables are set, and a few others on which they (especially self.s) depend.
+        self.initialised : Static variable to say if the REST interface is available and ready
+        self.base_url: The basic URL for the REST interface
+        self.s       : The python "requests" module session
+        self.headers : The format of information we prefer : json of course
+        Specification : https://www.nordugrid.org/arc/arc6/tech/rest/rest.html
+        """
+        if hasattr(self, "s"):
+            # Initialisation has already been successfully ran
+            return True
+        self.log.debug("Testing if the REST interface is available", "for %s" % self.ceHost)
+
+        result = self._prepareProxy()
+        if not result["OK"]:  # Probably everything is going to fail?
+            self.log.info("No proxy found -  Probably just no jobs to run for this CE?")
+            self.log.info("Setting REST interface false.")
+            return False
+
+        # Get the REST endpoint (service_url in ARC language) from the CS if available (preferred).
+        # It should be at
+        #   Resources/Sites/<Grid>/<Site>/CEs/<CE>/RESTEndpoint
+        # Otherwise query the CE for this value.
+
+        # Try getting service_url from the CS
+        ceString = cfgPath("Resources", "Sites", self.grid, self.site, "CEs", self.ceHost, "RESTEndpoint")
+        result = self._getFromCfgSvc(ceString)
+        if result["OK"]:
+            service_url = result["Value"]  # All well
+        if len(service_url) < 5:  # There is no endpoint in the CS. Discover it.
+            # The following command should expand to for example
+            # ldapsearch -x -LLL -h grendel.hec.lancs.ac.uk:2135 -b 'o=glue' GLUE2EndpointInterfaceName=org.nordugrid.arcrest GLUE2EndpointURL | grep GLUE2EndpointURL | awk -F ": " '{print $2}'
+            server = ldap3.Server("ldap://self.ceHost:2135")
```

Comment:
I guess this is a typo?
```suggestion
            server = ldap3.Server("ldap://" + self.ceHost + "2135")
```

Reply:
Thanks! Committed as it is.